src/compute-types/src/plan/flat_plan.rs

```diff
@@ -54,18 +69,11 @@ include!(concat!(
 ///      in the `nodes` map.
 ///  (3) Each [`LirId`] contained within `topological_order` is contained in the `nodes` map.
 ///
-/// Constrained graph structure:
-///
-///  (4) `LetRec` nodes only occur at the root or as inputs of other `LetRec` nodes.
-///  (5) Only `LetRec` nodes may introduce cycles into the plan.
-///  (6) Each input to a `LetRec` node is the root of an independent subplan.
-///  (7) `Let` nodes only occur at the root, as inputs of `LetRec` nodes, or as `body` inputs of
-///      other `Let` nodes.
-///  (8) Each input to a `Let` node is the root of an independent subplan.
```

Comment:
These invariants are still mostly true and partially relied upon by the code below. The "Preconditions" section replaces them, but there are still a couple references to the now-deleted invariants below, specifically to (6) (in `split_recursive`, (7) and (8) (in `split_lets`). I propose adding an invariant "the graphs structure forms a `valid_plan` as described in Preconditions"  and then referencing that in the comments below. Or alternatively, just remove the references in the comments below.

Reply:
Cool. The main thing I wanted to fix is that they are not invariants if they are not true.